test/parser/declarations/__snapshots__/functions.ts.snap

```diff
@@ -3015,9 +3015,9 @@ exports[`Declarations - Function > Declarations - Functions (fail) > function fo
 `;
 
 exports[`Declarations - Function > Declarations - Functions (fail) > function foo(p\\u0061ckage) { "use strict"; } 1`] = `
-"SyntaxError [1:37-1:38]: Unexpected strict mode reserved word
+"SyntaxError [1:43-1:44]: Unexpected strict mode reserved word
 > 1 | function foo(p\\u0061ckage) { "use strict"; }
-    |                                      ^ Unexpected strict mode reserved word"
+    |                                            ^ Unexpected strict mode reserved word"
```

Comment:
This seems incorrect(both original and new).

Reply:
This is tricky. The code set a flag (for this one, it's `Flags.HasStrictReserved` I think) when parsing `p\\u0061ckage`, the string literal could be valid for non-strict mode. Then it passes function body for "directives", and it triggers error by checking parser.flags.

Note we only set flag when parsing the function arguments, not saving any position information, it's impossible to get back to the real range for the failure.